polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -43,16 +43,39 @@ use sp_runtime::{
 	traits::{CheckedSub, Saturating},
 	RuntimeDebug,
 };
+use xcm::opaque::lts::{Junction::Parachain, MultiLocation, Parent};
+use xcm_executor::traits::ConvertLocation;
 
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Default, RuntimeDebug, TypeInfo)]
 pub struct ParaInfo<Account, Balance> {
 	/// The account that has placed a deposit for registering this para.
+	///
+	/// The given account ID is responsible for registering the code and initial head data, but may
+	/// only do so if it isn't yet registered. (After that, it's up to governance to do so.)
 	pub(crate) manager: Account,
-	/// The amount reserved by the `manager` account for the registration.
+	/// The total amount reserved for this para.
 	deposit: Balance,
 	/// Whether the para registration should be locked from being controlled by the manager.
 	/// None means the lock had not been explicitly set, and should be treated as false.
 	locked: Option<bool>,
+	/// The billing account of a parachain. This account is responsible for holding the required
+	/// deposit for this registered parachain.
+	///
+	/// This account will be responsible for covering all associated costs related to performing
+	/// parachain validation code upgrades.
+	///
+	/// When a parachain is newly registered, this will be set to the parachain manager. However,
+	/// at a later point, this can be updated by calling the
+	/// `set_parachain_billing_account_to_self` extrinsic.
+	///
+	/// None means that the billing account hasn't been explicitly set. In such a case, if the
+	/// parachain intends to schedule a code upgrade and is not a lease-holding parachain, either
```

Comment:
mildly confusing. Mentioning that "update will fail (if this is not explicit set)" would make it more clear.

Reply:
Agreed, I made this more clear.